### PR TITLE
[Doc][Serve] Add minimal docs for model wrappers and http adapters

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
This PR adds minimal documentation for model wrappers and http adapters. It mostly just surface existing docstrings. More end to end tutorials to be added. 

Preview link:
- Adapters: https://ray--23536.org.readthedocs.build/en/23536/serve/http-servehandle.html#http-adapters
- Wrappers: https://ray--23536.org.readthedocs.build/en/23536/ray-air/getting-started.html#serving

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/http-servehandle.rst [hunk 1]
```diff
@@ -152,6 +152,43 @@ To try it out, save a code snippet in a local python file (i.e. main.py) and in
     ray start --head
     python main.py
 
+.. _serve-http-adapters:
+
+HTTP Adapters
+^^^^^^^^^^^^^
+
+Ray Serve provides a suite of adapter for input schema conversion. You can directly
```
Comment (edoakes) [inline doc/source/serve/http-servehandle.rst:160]:
```suggestion
Ray Serve provides a suite of adapters to convert HTTP requests to ML inputs like `numpy` arrays. You can directly
```

File: doc/source/serve/http-servehandle.rst [hunk 2]
```diff
@@ -152,6 +152,43 @@ To try it out, save a code snippet in a local python file (i.e. main.py) and in
...
+import them and put them into your FastAPI app. Alternatively, just use it with
+:ref:`Ray AI Runtime (AIR) model wrapper<air-serve-integration>` feature to one click deploy pre-trained models.
```
Comment (edoakes) [inline doc/source/serve/http-servehandle.rst:162]: reverse the ordering of this: common path will be model wrappers, so lead with that

Comment (edoakes) [reply doc/source/serve/http-servehandle.rst:162]: same with the example please. first show how to use model wrapper, then fastapi API

File: python/ray/serve/model_wrappers.py [hunk 3]
```diff
@@ -39,6 +39,33 @@ def _load_predictor_cls(
 
 
 class ModelWrapper(SimpleSchemaIngress):
+    """Serve any Ray AIR predictor from a AIR checkpoint.
```
Comment (edoakes) [inline python/ray/serve/model_wrappers.py:42]:
```suggestion
    """Serve any Ray AIR predictor from an AIR checkpoint.
```

File: doc/source/serve/http-servehandle.rst [hunk 4]
```diff
@@ -152,6 +152,43 @@ To try it out, save a code snippet in a local python file (i.e. main.py) and in
...
+
+It has the following schema for input:
+
+.. _serve-ndarray-schema:
+
+.. autopydantic_model:: ray.serve.http_adapters.NdArray
+
+The input schema will automatically be part of the generated OpenAPI schema with FastAPI.
+
+Here is a list of adapters and please feel free to contribute more!
```
Comment (edoakes) [inline doc/source/serve/http-servehandle.rst:186]: add a call to action to file github issue

File: doc/source/serve/http-servehandle.rst [hunk 5]
```diff
@@ -152,6 +152,60 @@ To try it out, save a code snippet in a local python file (i.e. main.py) and in
...
+
+    from ray.serve.http_adapters import array_to_databatch
+    from ray.serve.model_wrappers import ModelWrapper
+    from ray import serve
+
+    serve.deployment(name="my_model")(ModelWrapper).deploy(
+        my_ray_air_predictor,
+        my_ray_air_checkpoint,
+        input_schema=array_to_databatch
+    )
```
Comment (edoakes) [inline doc/source/serve/http-servehandle.rst:179]: is there a way to do this without needing `serve.deployment()(Wrapper)`? This is not a common usage pattern, most users won't know that this works w/o understanding decorators.

File: doc/source/serve/http-servehandle.rst [hunk 6]
```diff
@@ -152,6 +152,60 @@ To try it out, save a code snippet in a local python file (i.e. main.py) and in
...
 
+.. _serve-http-adapters:
+
+HTTP Adapters
+^^^^^^^^^^^^^
+
+Ray Serve provides a suite of adapters to convert HTTP requests to ML inputs like `numpy` arrays.
+You can just use it with :ref:`Ray AI Runtime (AIR) model wrapper<air-serve-integration>` feature
+to one click deploy pre-trained models.
+Alternatively, you can directly import them and put them into your FastAPI app.
```
Comment (edoakes) [inline doc/source/serve/http-servehandle.rst:163]: can we link out to fastapi docs for details / a tutorial on how this works?

File: doc/source/serve/http-servehandle.rst [hunk 7]
```diff
@@ -152,6 +152,60 @@ To try it out, save a code snippet in a local python file (i.e. main.py) and in
...
+
+For example, we provide a simple adapter for n-dimensional array.
+
+With :ref:`model wrappers<air-serve-integration>`, you can specify it via the ``input_schema`` field.
+
+.. code-block:: python
+
+    from ray.serve.http_adapters import array_to_databatch
+    from ray.serve.model_wrappers import ModelWrapper
+    from ray import serve
```
Comment (edoakes) [inline doc/source/serve/http-servehandle.rst:173]: nit: reverse order (top-level serve import first)

File: doc/source/serve/http-servehandle.rst (hunk 6 above)
Comment (edoakes) [reply doc/source/serve/http-servehandle.rst:163]: nevermind, addressed below.

File: doc/source/serve/http-servehandle.rst [hunk 8]
```diff
@@ -152,6 +152,60 @@ To try it out, save a code snippet in a local python file (i.e. main.py) and in
...
+
+    serve.deployment(name="my_model")(ModelWrapper).deploy(
+        my_ray_air_predictor,
+        my_ray_air_checkpoint,
+        input_schema=array_to_databatch
+    )
+
+You can also bring the adapter to your own FastAPI app using
+`Depends <https://fastapi.tiangolo.com/tutorial/dependencies/#import-depends>`_.
+
```
Comment (edoakes) [inline doc/source/serve/http-servehandle.rst:183]: please also mention that you can check out the openapi docs to see the input format!

File: doc/source/serve/http-servehandle.rst (hunk 5 above)
Comment (simon-mo) [reply doc/source/serve/http-servehandle.rst:179]: Yes but not at the moment. It's a much bigger change, requiring changing tests and others.

File: doc/source/serve/http-servehandle.rst (hunk 8 above)
Comment (simon-mo) [reply doc/source/serve/http-servehandle.rst:183]: mentioned below :D

File: python/ray/serve/tests/test_model_wrappers.py [hunk 9]
```diff
@@ -34,17 +39,7 @@ def predict(self, data: DataBatchType) -> DataBatchType:
...
-
-    @classmethod
-    def from_dict(cls, data: dict) -> "Checkpoint":
-        return cls(data["increment"])
-
-    @classmethod
-    def from_uri(cls, uri: str) -> "Checkpoint":
-        with open(uri) as f:
-            return cls(json.load(f))
+    pass
```
Comment (simon-mo) [inline python/ray/serve/tests/test_model_wrappers.py:42]: Checkpoint already had these method

File: doc/source/serve/http-servehandle.rst [hunk 10]
```diff
@@ -152,6 +152,60 @@ To try it out, save a code snippet in a local python file (i.e. main.py) and in
...
+    from ray import serve
+    from ray.serve.http_adapters import json_to_ndarray
+    from ray.serve.model_wrappers import ModelWrapperDeployment
+
+    ModelWrapperDeployment.options(name="my_model").deploy(
+        my_ray_air_predictor,
+        my_ray_air_checkpoint,
+        input_schema=json_to_ndarray
+    )
```
Comment (edoakes) [inline doc/source/serve/http-servehandle.rst:179]: much better! :)

**Review by edoakes — APPROVED** [review]
Looks much nicer, thanks for following up on the polish items.

**Pull request merged**